### PR TITLE
HandlersTimeProfiler: add an example notebook showing usage

### PR DESCRIPTION
**Pull request by harsh8398**
Completes #1427

Description:

- Added example notebook showing usage of HandlerTimeProfiler
- Added notebook entry in README as well

### CODE REVIEW

**Comment by harsh8398** [comment]
Just want to add that, while modifying README I noticed that pre-commit was suggesting lot of whitespace related changes in the README.

**Comment by vfdev-5** [comment]
@harsh8398 thanks for the PR ! I was wondering if you also see the cell with printed results like below:
Screen Shot 2020-11-22 at 15 18 42

> Just want to add that, while modifying README I noticed that pre-commit was suggesting lot of whitespace related changes in the README.

About the precommit, I think it should be related to https://github.com/pytorch/ignite/pull/1468. Could you please dump the diff once with the precommit, maybe it is OK and just accept them ?

**Comment by harsh8398** [comment]
@vfdev-5  was able to fix that spill by using the trick mentioned here. The output looks as follows now:
image

> Could you please dump the diff once with the precommit, maybe it is OK and just accept them ?

Sorry, I didn't get you. What are you suggesting? Do you want me to add those precommit changes in this PR?

**Comment by vfdev-5** [comment]
@sdesrozis could you please review the PR and merge if it is OK

**Comment by sdesrozis** [comment]
@harsh8398 Maybe we should add a line to install `Pytorch-ignite` in notebook
```
!pip install --upgrade --pre pytorch-ignite
```
What do you think ?

There is a warning from `ignite/contrib/handlers/time_profilers.py:587`
```
UserWarning: To copy construct from a tensor, it is recommended to use sourceTensor.clone().detach() or sourceTensor.clone().detach().requires_grad_(True), rather than torch.tensor(sourceTensor).
  data = torch.tensor(data, dtype=torch.float32)
```
Could we remove it ?

**Comment by harsh8398** [comment]
@sdesrozis I have added the suggested cell at the top. Also, I've raised a PR(https://github.com/pytorch/ignite/pull/1487) for the change required to remove the warning.

File: examples/notebooks/HandlersTimeProfiler_MNIST.ipynb [hunk 1]
```diff
@@ -9,6 +9,37 @@
...
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "/home/perceptron/.pyenv/versions/ignite/bin/python\r\n"
+     ]
+    }
+   ],
+   "source": [
+    "!which python"
```
Comment (sdesrozis) [inline examples/notebooks/HandlersTimeProfiler_MNIST.ipynb:33]: @harsh8398 I think there is a problem here. The line should be 
```
!pip install --pre pytorch-ignite
```

**Comment by sdesrozis** [comment]
@harsh8398 I think you didn't commit the good installation command line in notebook. Could you check and fix please ?

**Comment by harsh8398** [comment]
@sdesrozis Sorry for that, Its fixed now.

**Comment by sdesrozis** [comment]
@harsh8398 Thank you LGTM !

**Review by sdesrozis — APPROVED** [review]
Thanks

**Pull request merged**